scala-package/macros/src/main/scala/org/apache/mxnet/NDArrayMacro.scala

```diff
@@ -129,6 +119,8 @@ private[mxnet] object NDArrayMacro {
         }
         impl += base
       })
+      // add default out parameter
+      argDef += "out : Option[NDArray] = None"
```

Comment:
is this to keep it the same as Python? What is the use of this parameter?

Reply:
The usage is to save memory space, you pass in a pointer and C will store the value in there...